desc/integrals/bounce_integral.py

```diff
@@ -1,26 +1,729 @@
+    systems such that the function to approximate is periodic. The Chebyshev
+    polynomials are preferred to other orthogonal polynomial series since
+    fast discrete polynomial transforms (DPT) are implemented via fast transform
+    to Chebyshev then DCT. Although nothing prohibits a direct DPT, we want to
+    rely on existing libraries. There are other reasons to prefer Chebyshev series
+    not discussed here. Therefore, a Fourier-Chebyshev series is chosen to
+    interpolate θ(α,ζ), and a piecewise Chebyshev series interpolates |B|(ζ).
+
+    Computing accurate series expansions in (α, ζ) coordinates demands
+    particular interpolation points in that coordinate system. Newton iteration
+    is used to compute θ at these points. Note that interpolation is necessary
+    because there is no transformation that converts series coefficients in
+    periodic coordinates, e.g. (ϑ, ϕ), to a low order polynomial basis in
+    non-periodic coordinates. For example, one can obtain series coefficients in
+    (α, ϕ) coordinates from those in (ϑ, ϕ) as follows
+        g : ϑ, ϕ ↦ ∑ₘₙ aₘₙ exp(j [mϑ + nϕ])
+        g : α, ϕ ↦ ∑ₘₙ aₘₙ exp(j [mα + (m ι + n)ϕ])
+    However, the basis for the latter are trigonometric functions with
+    irrational frequencies, courtesy of the irrational rotational transform.
+    Globally convergent root-finding schemes for that basis (at fixed α) are
+    not known. The denominator of a close rational could be absorbed into the
+    coordinate ϕ, but this balloons the frequency, and hence the degree of the
+    series.
+
+    Recall that periodicity enables faster convergence, motivating the desire
+    to instead interpolate |B|(ϑ, ϕ) with a double Fourier series and applying
+    bisection methods to find bounce points with mesh size inversely
+    proportional to the max frequency along the field line: M ι + N. ``Bounce2D``
+    does not use that approach as that root-finding scheme is inferior.
+    The reason θ is not interpolated with a double Fourier series θ(ϑ, ζ) is
+    because quadrature points along |B|(α=α₀, ζ) can be identified by a single
+    variable; evaluating the multivariable map θ(ϑ(α, ζ), ζ) is expensive
+    compared to evaluating the single variable map θ(α=α₀, ζ).
+    Another option is to use a filtered Fourier series,
+    doi.org/10.1016/j.aml.2006.10.001.
+
+    After computing the bounce points, the supplied quadrature is performed.
+    By default, this is a Gauss quadrature after removing the singularity.
+    Fast fourier transforms interpolate smooth functions in the integrand to the
+    quadrature nodes. Quadrature is chosen over Runge-Kutta methods of the form
+        ∂Fᵢ/∂ζ =f(λ,ζ,{Gⱼ}) subject to Fᵢ(ζ₁) = 0
+    because a fourth order Runge-Kutta method is equivalent to a quadrature
+    with Simpson's rule. Our quadratures resolve these integrals more
+    efficiently, and the fixed nature of quadrature performs better on GPUs.
```

Comment:
FYI, Neo uses a fourth order Runge-Kutta. That method would fail for strongly singular integrals such as the ones in the Gammas. Stellopt uses tanh-sinh quadrature for that, which we tested performed poorly in our implementation.

Reply:
Stellopt uses tanh-sinh quadrature for Gamma, which we tested performed poorly in our implementation.